lib/presentation/widget/upload_file/upload_file_viewmodel.dart

```diff
@@ -128,45 +141,70 @@ class UploadFileViewModel extends BaseViewModel {
 
   void _handleUploadAndShare() {
     if (_fileInfoArgument != null) {
-      store.dispatch(uploadFileAction(_fileInfoArgument));
+      store.dispatch(uploadAndShareFileAction(_fileInfoArgument));
+    }
+  }
+
+  Future<List<AutoCompleteResult>> getAutoCompleteSharing(
+      String pattern) async {
+    return await _getAutoCompleteSharingInteractor
+        .execute(AutoCompletePattern(pattern), AutoCompleteType.sharing)
+        .then(
+          (viewState) => viewState.fold(
+            (failure) => <AutoCompleteResult>[],
+            (success) => (success as AutoCompleteViewState).results,
+          ),
+        );
+  }
+
+  String getSuggestionDisplayName(AutoCompleteResult autoCompleteResult) {
+    if (autoCompleteResult is UserAutoCompleteResult) {
+      return autoCompleteResult.fullName().isNotEmpty
+          ? autoCompleteResult.fullName()
+          : autoCompleteResult.display;
+    }
+    return autoCompleteResult.display;
```

Comment:
What do you think about returning one ternary ? I find that more readable but I don't mind